narwhal/primary/src/synchronizer.rs

```diff
@@ -113,7 +117,12 @@ impl Inner {
         Ok(())
     }
 
-    async fn accept_certificate_internal(&self, certificate: Certificate) -> DagResult<()> {
+    // State lock must be held when calling this function.
+    async fn accept_certificate_internal(
+        &self,
+        _lock: &MutexGuard<'_, State>,
```

Comment:
IIUC the introduced changes on the current code will improve the safety and semantics - but we don't expect to fix/reveal something from the existing codebase as we already acquire the necessary locks, right?

Reply:
Yes, this is just to help us confirm that these functions are called with locks. No locking logic is changed in this PR.